### PR TITLE
Version 1.3.17

### LINKED ISSUE #59: Allow sienaDataCreate to accept existing lists as objects
As discussed with @TomSnijders on 2023-01-06

### PR DESCRIPTION
**Pull request by TomSnijders**
# Description

* List is allowed as input for sienaDataCreate; and missings are allowed for simulated values in sienaGOF.

* Closed #59 by allowing list as input for sienaDataCreate

# Checklist:

## Checks

- [ X] The package builds on my OS without issues
- [ X] My changes generate no new warnings

## Documentation

- [X] I have added a description of all changes to this pull request above and to the NEWS.md file
- [X] I have bumped the version in the DESCRIPTION and man/RSiena-package.md files by the appropriate increment:
  - 'major' (incompatible API changes)
  - 'minor' (add functionality in a backwards compatible manner)
  - 'patch' (backwards compatible bug fixes)

### CODE REVIEW

File: R/sienaDataCreate.r [hunk 1]
```diff
@@ -235,7 +235,22 @@ sienaDataCreate<- function(..., nodeSets=NULL, getDocumentation=FALSE)
...
+		stop('need some objects')
+	}
+	if (length(dots) == 1)
+	{
+		ldots <- list(...)
+		dotsIsList <- (is.list(ldots[[1]]))
+# If dotsIsList, it needs to be a list of variables
+		if (dotsIsList) 
+		{
+			dots <- as.list(substitute(...))[-1]
```
Comment (jhollway) [inline R/sienaDataCreate.r:247]: Why is `dots` overwritten here? How does it impact e.g. `nm ← names(dots)` below?

File: man/sienaDataCreate.Rd [hunk 2]
```diff
@@ -23,16 +25,16 @@ sienaDataCreate(..., nodeSets=NULL, getDocumentation=FALSE)
...
 data. If there is more than one nodeSet they must all be specified.
 Function \code{\link{print01Report}} will give a basic description of the data
 object and is a check useful, e.g., for diagnosing problems.
 }
 \value{
 An object of class "siena" which is designed to be used in a siena
-model fit. The components of the object are.
-\item{nodeSets}{List of node sets involved}
+model fit by \code{link{siena07}}. The components of the object are:
+\item{nodeSets}{List of node setss}
```
Comment (jhollway) [inline man/sienaDataCreate.Rd:37]: typo.

File: man/sienaDataCreate.Rd [hunk 3]
```diff
@@ -55,6 +57,8 @@ model fit. The components of the object are.
 mynet <- sienaDependent(array(c(s501, s502, s503), dim=c(50, 50, 3)))
 mybeh <- sienaDependent(s50a, type="behavior")
 mydata <- sienaDataCreate(mynet, mybeh)
+# This gives the same result as
+mydata <- sienaDataCreate(list(mynet, mybeh))
```
Comment (jhollway) [inline man/sienaDataCreate.Rd:61]: Excellent, I'll test this next week or so.

**Review by jhollway — APPROVED** [review]
Thanks Tom, looks good. I'll test how the sienaDataCreate() list passing works with different configurations later next week after it's been build on CRAN.

File: man/sienaDataCreate.Rd (hunk 2 above)
Comment (TomSnijders) [reply man/sienaDataCreate.Rd]: typo corrected

**Pull request merged**